lib/splicer/dns_made_easy/provider.rb

```diff
@@ -64,19 +64,30 @@ def create_record_in_zone(record, zone)
         create_record(domain.id, record)
       end
 
+      def get_records_for(zone)
+        Splicer.logger.debug "[SPLICER][DNSMADEEASY] #get_records_for zone=#{zone.inspect}"
+        domain = find_domain(zone.name)
+        return false unless domain.persisted?
+        fetch_records(domain.id)
+      end
+
       private
 
       # @param [Integer] domain_id
       # @return [Array<Splicer::DnsMadeEasy::Record>]
       def fetch_records(domain_id)
-        response = JSON.parse(client.get(record_url(domain_id)))
-        list = []
-        response['data'].each do |r|
-          list << Record.new(r)
+        records_call = client.get(record_url(domain_id))
+        records = JSON.parse(records_call)['data']
+
+        @records = []
+        records.each do |record|
+          new_record = Record.new
+          record.each {|attribute, type| new_record.add_attribute(attribute, type)}
```

Comment:
Since you have `attr_accessor` you can use `record.each {|attribute, type| new_record.send(:"#{attribute}=", type)}` instead of `record.each {|attribute, type| new_record.add_attribute(attribute, type)}`